lib/graphql/batch/loader.rb

```diff
@@ -13,9 +15,12 @@ def self.load_many(keys)
       self.for.load_many(keys)
     end
 
+    attr_accessor :loader_key
+
     def load(key)
-      if queue.frozen?
-        raise "Loader can't be used after batch load"
+      loader = Executor.current.loaders[loader_key] ||= self
+      if loader != self
+        raise "load called on loader that wasn't registered with executor"
```

Comment:
It could happen if you are mixing up re-using a loader and getting a loader using Loader.for, which could happen from using a loader in another thread, but could also happen in a single thread:

```ruby
  def test_load_on_different_loader
    loader = EchoLoader.for
    assert_equal :a, loader.load(:a).sync
    loader2 = EchoLoader.for
    promise = loader2.load(:b)
    err = assert_raises(RuntimeError) do
      loader.load(:c).sync
    end
    assert_equal "load called on loader that wasn't registered with executor", err.message
  end
```